homeassistant/components/knx/fan.py

```diff
@@ -32,18 +42,22 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 class KNXFan(KnxEntity, FanEntity):
     """Representation of a KNX fan."""
 
-    def __init__(self, device: XknxFan):
+    def __init__(self, device: XknxFan) -> None:
         """Initialize of KNX fan."""
+        self._device: XknxFan
         super().__init__(device)
 
-        if self._device.mode == FanSpeedMode.STEP:
+        self._step_range: tuple[int, int] | None
+        if device.max_step:
+            # FanSpeedMode.STEP:
             self._step_range = (1, device.max_step)
         else:
+            # FanSpeedMode.PERCENT
```

Comment:
This else is now obsolete, you can set the default on line 50:

```py
self._step_range: tuple[int, int] | None = None
```

Reply:
alright. Thanks!

Reply:
👍  

As a general rule of thumb though (As I've seen the pattern multiple times in this integration). An `else` is almost never needed... 😉